packages/core/src/profile/link.ts

```diff
@@ -0,0 +1,86 @@
+import { parseKey } from '@preevy/common'
+import * as jose from 'jose'
+import crypto from 'crypto'
+import fetch from 'node-fetch'
+import { TokenExpiredError, TokesFileSchema, getTokens } from '../login'
+import { profileStore } from './store'
+import { Store } from '../store'
+import { Logger } from '../log'
+
+export type Org = {id: string; name: string; role: string}
+
+export const link = async (
+  store: Store,
+  dataDir: string,
+  lcUrl: string,
+  logger: Logger,
+  promptUserWithChooseOrg: (orgs: Org[]) => Promise<Org>
+) => {
+  let tokens: TokesFileSchema | undefined
+  try {
+    tokens = await getTokens(dataDir)
+  } catch (e) {
+    if (e instanceof TokenExpiredError) {
+      logger.info('Session is expired, please log in again')
+      return
+    }
+    throw e
+  }
+
+  if (tokens === undefined) {
+    logger.info('Please log in to link profile')
+    return
+  }
+
+  const orgsResponse = await fetch(
+    `${lcUrl}/orgs`,
+    { headers: { 'Content-Type': 'application/json', Authorization: `Bearer ${tokens.access_token}` } }
+  )
+
+  if (!orgsResponse.ok) throw new Error(`Could not fetch orgs from Livecycle API. ${orgsResponse.status}: ${orgsResponse.statusText}`)
+
+  const orgs = await orgsResponse.json() as Org[]
+
+  let chosenOrg: Org
```

Comment:
style nit - wrap in function `chooseOrg(orgs, promptUserWithChooseOrg)` to shorten and avoid `let`